src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,58 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+import com.intellij.psi.tree.TokenSet
+
+/**
+ * Mimics [com.intellij.formatting.alignment.AlignmentStrategy], but offers more flexible API
+ */
+interface RustAlignmentStrategy {
+    /**
+     * Requests current strategy for alignment to use for given child.
+     */
+    fun getAlignment(child: ASTNode, parent: ASTNode?): Alignment?
+
+    companion object {
+        /**
+         * Always returns `null`.
+         */
+        val NULL_STRATEGY = object : RustAlignmentStrategy {
+            override fun getAlignment(child: ASTNode, parent: ASTNode?): Alignment? = null
+        }
```

Comment:
Instead of

```
class Foo {
    companion object {
        val QUUX = object I {}
    }
}
```

you can simply

```
class Foo {
    object QUUX: I {}
}
```